DogTales/src/components/physics.cpp

```diff
@@ -0,0 +1,29 @@
+#include <src/components/physics.hpp>
+
+namespace component {
+
+Physics::Physics(glm::vec2 friction, float gravity, float mass) : m_friction(friction), m_gravity(gravity), m_mass(mass) {}
+
+void Physics::tick(bave::Seconds const dt) {
+	static constexpr bave::Seconds ft{0.005}; // 5ms
+
+	if (dt.count() > tick_limit) { return; } // return for unexpected dt values, particularly during the beginning of the state
+
+	for (bave::Seconds t{}; t <= dt; t += ft) {
+		integrate(t.count());
+	}
+}
```

Comment:
would this fix work?
`for (bave::Seconds t{ m_accumulator }; t <= dt; t += ft) {
	integrate(t);
	if (t + ft >= dt) { m_accumulator = dt - t; }
}`

Reply:
I don't think so, something like this would be better:
```cpp
void Physics::tick(bave::Seconds dt) {
	// ...
	for (dt += m_residue; dt > ft; dt -= ft) {
		integrate(ft);
	}
	m_residue = dt;
}
```